pkg/syncer/metadatasyncer.go

```diff
@@ -2213,6 +2214,77 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 				log.Errorf("PVUpdated: QueryVolume failed for volume %q with err=%+v", oldPv.Spec.CSI.VolumeHandle, err.Error())
 				return
 			}
+			if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.PodVMOnStretchedSupervisor) {
+				cnsVolumeInfoCRList := &cnsvolumeinfov1alpha1.CNSVolumeInfoList{}
+				err := metadataSyncer.cnsOperatorClient.List(ctx, cnsVolumeInfoCRList,
+					client.InNamespace(common.GetCSINamespace()))
+				if err != nil {
+					log.Errorf("failed to list %q CR. Error: %+v",
+						cnsvolumeinfo.CRDSingular, err)
+					return
+				}
+				foundCnsVolumeInfoCR := false
+				cnsVolumeInfoCR := &cnsvolumeinfov1alpha1.CNSVolumeInfo{}
+				if len(cnsVolumeInfoCRList.Items) > 0 {
+					for _, item := range cnsVolumeInfoCRList.Items {
+						if item.Spec.VolumeID == newPv.Spec.CSI.VolumeHandle {
+							foundCnsVolumeInfoCR = true
+							cnsVolumeInfoCR = item.DeepCopy()
+							break
+						}
+					}
+				}
+				if foundCnsVolumeInfoCR {
+					namespace := cnsVolumeInfoCR.Spec.Namespace
+					storagePolicyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+					err = metadataSyncer.cnsOperatorClient.List(ctx, storagePolicyUsageList, client.InNamespace(namespace))
+					if err != nil {
+						log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+							storagepolicyusagev1alpha1.CRDSingular, namespace, err)
+						return
+					}
+					foundStoragePolicyUsageCR := false
+					oldStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+					newStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+					if len(storagePolicyUsageList.Items) > 0 {
+						for _, item := range storagePolicyUsageList.Items {
+							policyId := cnsVolumeInfoCR.Spec.StoragePolicyID
+							scName := cnsVolumeInfoCR.Spec.StorageClassName
+							if item.Spec.StoragePolicyId == policyId && item.Spec.StorageClassName == scName {
+								log.Debugf("Found storagePolicyUsage CR with matching "+
+									"storagePolicyId: %q & storageClassName: %q in namespace: %q", policyId, scName,
+									namespace)
+								oldStoragePolicyUsageCR = item.DeepCopy()
+								newStoragePolicyUsageCR = item.DeepCopy()
+								foundStoragePolicyUsageCR = true
+								break
+							}
+						}
+					}
+					if foundStoragePolicyUsageCR {
+						newStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used.Add(
+							*resource.NewQuantity(int64(newPv.Spec.Size()),
+								oldStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Format))
+						err := patchStoragePolicyUsage(ctx, metadataSyncer.cnsOperatorClient, oldStoragePolicyUsageCR,
```

Comment:
I'm thinking about this again. I think we still need to do two patches here:
1) patch an increase of "reserved" in storagePolicyUsageCR.
2) patch a decrease of "reserved" and increase of "used in storagePolicyUsageCR.
This way it is consistent with how dynamically provisioning is handled.
The quota controller updates StoragePolicyQuota based on StoragePolicyUsage. If it only rely on "used" to update both "reserved" and "used", it may not be accurate.

Reply:
The latest commit/changes should address this comment. Will add the test results in the PR description